### PR TITLE
Cleaning up site template & theme updates.

### PR DESCRIPTION
**Pull request by parkr**
- [x] Theme template example should link to the post, not just quote it
- [x] Add `.gitignore` for `.bundle`, `Gemfile.lock`, `.sass-cache/`, etc.
-  ~~Maybe lock gemspec to 3.2 or greater?~~ Just use whatever version of Jekyll the user is running
- [x] Migrate existing site template to use a theme

/cc @jekyll/ecosystem, anything else you can think of to improve this? Check out the current incarnation in Jekyll v3.2.0.pre.beta1. 😄  Just run `jekyll new-theme <theme-name>` and see what's inside!

### CODE REVIEW

**Comment by envygeeks** [comment]
Where are the files going?

**Comment by SuriyaaKudoIsc** [comment]
Doesn't work: #4928!

**Comment by parkr** [comment]
@SuriyaaKudoIsc #4928 is for our site, jekyllrb.com. This pull request is for our theme template for new themes.

**Comment by SuriyaaKudoIsc** [comment]
Oh, I see.

**Comment by parkr** [comment]
This is ready. Can I have some 👀 on it? @benbalter, especially. ❤️

File: lib/theme_template/example/index.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
 Lorem ipsum dolor sit amet, quo id prima corrumpit pertinacia, id ius dolor dolores, an veri pertinax explicari mea. Agam solum et qui, his id ludus graeco adipiscing. Duis theophrastus nam in, at his vidisse atomorum. Tantas gloriatur scripserit ne eos. Est wisi tempor habemus at, ei graeco dissentiet eos. Ne usu aliquip sanctus conceptam, te vis ignota animal, modus latine contentiones ius te.
```
Comment (benbalter) [inline lib/theme_template/example/index.html]: Should the index have a `default` or `page` layout?

Comment (parkr) [inline lib/theme_template/example/index.html]: I have a comment above that says `# Specify a layout from your theme!` so users don't feel confined to `default` or `page` or whatever. Should we just choose one anyway?

Comment (benbalter) [inline lib/theme_template/example/index.html]: I'm just thinking that we're initializing it with a theme, but when you view the site, it'll be unthemed out of the box.

Comment (parkr) [inline lib/theme_template/example/index.html]: Done.

**Comment by parkr** [comment]
@jekyllbot: merge +minor

**Pull request merged**